hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -1335,10 +1335,14 @@ public boolean isClusteringEnabled() {
     return inlineClusteringEnabled() || isAsyncClusteringEnabled();
   }
 
-  public boolean isRollbackPendingClustering() {
+  public boolean isRollbackPendingClusteringOnConflict() {
     return getBoolean(HoodieClusteringConfig.ROLLBACK_PENDING_CLUSTERING_ON_CONFLICT);
   }
 
+  public boolean isRollbackPendingClustering() {
+    return getBoolean(HoodieClusteringConfig.ROLLBACK_PENDING_CLUSTERING);
+  }
+
```

Comment:
Could you avoid changing the logic of existing APIs and rather adding a new API named `isRollbackPendingClusteringWhenDisablingClustering`?

Reply:
the older getter did not have the right name and so took the liberty to fix it. These are writeConfig apis. we are not changing the config key as such, just the getters in the writeConfig. So, I guess we should be good.

Reply:
infact, the older config key is `hoodie.clustering.rollback.pending.replacecommit.on.conflict`, just the getter wasn't properly named.

Reply:
Note that `withRollbackPendingClustering()` and `isRollbackPendingClustering()` are public methods.  Any code outside Hudi repo using these two APIs will be affected by this implicit semantics change.  If you insist on the changes, let's create a documentation update to call it out.  But I would rather keep the naming the same to avoid any behavior change for the same APIs and focus on better naming for any new APIs.